src/main/java/seedu/address/model/event/EventName.java

```diff
@@ -0,0 +1,59 @@
+package seedu.address.model.event;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Event's name in the Scheduler.
+ * Guarantees: immutable; is valid as declared in {@link #isValidName(String)}
+ */
+public class EventName {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+    private final String name;
```

Comment:
We should do boundary value testing for this in the future

Reply:
Ok noted.